Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace Unity.MeshSync {
+    partial class BaseMeshSync {
+#if UNITY_EDITOR
+        /// <summary>
+        /// Names of all maps that could be baked to a render texture and might need to be saved to the asset database.
+        /// </summary>
+        static readonly string[] textureNames = {
+            MeshSyncConstants._MetallicGlossMap,
+            MeshSyncConstants._BaseMap,
+            MeshSyncConstants._MaskMap
+        };
+
+        private List<Tuple<Material, string>> pendingMaterialUpdates = new List<Tuple<Material, string>>();
+
+        /// <summary>
+        /// Returns the path where the given texture should be saved.
+        /// </summary>
+        /// <param name="mat">Material that owns the texture</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private string GetSavePath(Material mat, string textureName) {
+            return Path.Combine(GetAssetsFolder(), $"{mat.name}_{textureName}_baked.png");
+        }
+
+        /// <summary>
+        /// Saves any render targets to the AssetDatabase as textures.
+        /// </summary>
+        private void SaveMaterialRenderTexturesToAssetDatabase() {
+            // Check if there are render textures that need to be saved to the asset database:
+            foreach (MaterialHolder materialHolder in materialList) {
+                foreach (string textureName in textureNames) {
+                    var mat = materialHolder.material;
+
+                    if (mat == null || !mat.HasProperty(textureName)) {
+                        continue;
+                    }
+
+                    var tex = mat.GetTexture(textureName);
+                    if (tex == null) {
+                        continue;
+                    }
+
+                    if (tex is RenderTexture renderTarget) {
+                        using (var texture = new Texture2DDisposable(
+                                   new Texture2D(renderTarget.width, renderTarget.height,
+                                   UnityEngine.TextureFormat.RGBA32, true))) {
```

Comment:
Not a blocker, but probably worth adding a Texture2DDisposable constructor that takes the parameters for the Texture2D so you don't need to worry about creating a `new Texture2d()` here too